src/app/zap-templates/templates/app/CHIPClientCallbacks-src.zapt

```diff
@@ -340,50 +339,51 @@ bool emberAfDiscoverCommandsReceivedResponseCallback(ClusterId clusterId, uint16
 {{#if isList}}
 void {{asUpperCamelCase parent.name}}Cluster{{asUpperCamelCase name}}ListAttributeFilter(TLV::TLVReader * tlvData, Callback::Cancelable * onSuccessCallback, Callback::Cancelable * onFailureCallback)
 {
-    // TODO: Add actual support for array and lists.
-    const uint8_t * message = nullptr;
-    uint16_t messageLen     = 0;
-    EmberAfStatus res = PrepareListFromTLV(tlvData, message, messageLen);
-    if (res != EMBER_ZCL_STATUS_SUCCESS)
+    CHIP_ERROR err = CHIP_NO_ERROR;
+    size_t count = 0;
+    TLV::TLVType tmpType;
+    TLV::TLVReader reader;
+
+    if (tlvData != nullptr && tlvData->GetType() == chip::TLV::TLVType::kTLVType_Array)
+    {
+        reader.Init(*tlvData);
+        reader.EnterContainer(tmpType);
+    }
+    if (tlvData == nullptr || tlvData->GetType() != chip::TLV::TLVType::kTLVType_Array ||
+        chip::TLV::Utilities::Count(reader, count, false /* recursive */) != CHIP_NO_ERROR || !chip::CanCastTo<uint16_t>(count))
     {
         Callback::Callback<DefaultFailureCallback> * cb = Callback::Callback<DefaultFailureCallback>::FromCancelable(onFailureCallback);
-        cb->mCall(cb->mContext, res);
+        cb->mCall(cb->mContext, EMBER_ZCL_STATUS_INVALID_VALUE);
```

Comment:
Since you are using `VerifyOrExit` and `SuccessOrExit` macro later, can you reformat the code so it does something like: 
```
VerifyOrExit(tlvData != nullptr && tlvData->GetType() == chip::TLV::TLVType::kTLVType_Array, err = EMBER_ZCL_STATUS_INVALID_VALUE);

reader.Init(*tlvData);
reader.EnterContainer(tmpType);
SuccessOrExit(chip::TLV::Utilities::Count(reader, count, false /* recursive */));
```